etc/testing/circle/workloads/go-test-results/collector/main.go

```diff
@@ -54,52 +51,144 @@ func main() {
 	if len(resultsFolder) == 0 {
 		logger.Fatal("TEST_RESULTS needs to be populated to find the test results folder.")
 	}
-	var branchFolderName string
-	if branch := sanitizeName(os.Getenv("CIRCLE_BRANCH")); len(branch) != 0 {
-		branchFolderName = branch
-	} else if tag := sanitizeName(os.Getenv("CIRCLE_TAG")); len(tag) != 0 {
-		branchFolderName = tag
-	} else {
-		logger.Fatal("CIRCLE_BRANCH or CIRCLE_TAG needs to be populated to upload test results.")
-	}
-	jobName := sanitizeName(os.Getenv("CIRCLE_JOB"))
-	if len(jobName) == 0 {
-		logger.Fatal("CIRCLE_JOB needs to be populated to upload test results.")
+	if _, err := os.Stat(resultsFolder); os.IsNotExist(err) {
+		logger.Fatalf("The test result folder at %v does not exist. Exiting early.", resultsFolder)
 	}
-
+	// connect and authenticate to pachyderm
 	pachClient, err := client.NewFromURIContext(context.Background(), pachdAddress)
 	if err != nil {
 		logger.Fatalf("Provisioning pach client: %v", err)
 	}
 	pachClient.SetAuthToken(robotToken)
-
+	// retry in case the parent commit is still in progress from another pipeline
+	var commit *pfs.Commit
+	for i := 0; i < commitMaxRetries; i++ {
+		commit, err = pachClient.StartProjectCommit(projectName, repoName, "master")
+		if err != nil {
+			logger.Printf("Unable to start commit due to error. Retrying. Error: %v", err)
+			time.Sleep(commitRetryBackoff)
+		} else {
+			break
+		}
+	}
+	if err != nil {
+		logger.Fatalf("Starting commit: %v", err)
+	}
+	defer func() {
+		if err = pachClient.FinishProjectCommit(projectName, repoName, "master", commit.GetID()); err != nil {
+			logger.Fatalf("Finishing commit: %v", err)
+		}
+	}()
+	// upload general job information
+	jobInfo, err := findJobInfoFromCI()
+	if err != nil {
+		logger.Fatalf("Collecting job info: %v", err)
+	}
+	basePath := findBasePath(jobInfo)
+	if err = uploadJobInfo(basePath, jobInfo, pachClient, commit); err != nil {
+		logger.Fatalf("Uploading job info: %v", err)
```

Comment:
For log messages, I'd err on the side of indicating a problem.  For errors, this is a good message, for log messages, I'd be explicit: "problem uploading job info".